src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1006,12 +1006,55 @@ public override IOperation DefaultVisit(IOperation operation, int? captureIdForR
             throw ExceptionUtilities.Unreachable;
         }
 
-        #region PROTOTYPE(dataflow): Naive implementation that simply clones nodes and erases SemanticModel, likely to change
         internal override IOperation VisitNoneOperation(IOperation operation, int? captureIdForResult)
         {
-            return Operation.CreateOperationNone(semanticModel: null, operation.Syntax, operation.ConstantValue, () => VisitArray(operation.Children.ToImmutableArray()), operation.IsImplicit);
+            if (_currentStatement == operation)
+            {
+                VisitNoneOperationStatement(operation);
+                return null;
+            }
+            else
+            {
+                return VisitNoneOperationExpression(operation, captureIdForResult);
+            }
+        }
+
+        private void VisitNoneOperationStatement(IOperation operation)
+        {
+            Debug.Assert(_currentStatement == operation);
+            foreach (var child in operation.Children)
+            {
+                VisitStatement(child);
+            }
+        }
+
+        private IOperation VisitNoneOperationExpression(IOperation operation, int? captureIdForResult)
+        {
+            int numChildren = 0;
+            foreach (var child in operation.Children)
+            {
+                numChildren++;
```

Comment:
>numChildren++ [](start = 16, length = 13)

Alternatively we can simply store the current size of the ```_evalStack``` and then pop according to that. No need to increment inside the loop